src/helper.cc

```diff
@@ -1561,30 +1559,25 @@ Helper::Session::checkForTimedOutRequests(bool const retry)
         requestsIndex.erase(it);
         requests.pop_front();
         debugs(84, 2, "Request " << r->request.Id << " timed-out, remove it from queue");
-        void *cbdata;
-        bool retried = false;
         if (retry && r->request.retries < MAX_RETRIES && cbdataReferenceValid(r->request.data)) {
             debugs(84, 2, "Retry request " << r->request.Id);
             ++r->request.retries;
             parent->submitRequest(r);
-            retried = true;
-        } else if (cbdataReferenceValidDone(r->request.data, &cbdata)) {
```

Comment:
Same as in other change requests, let's restore the original logic here to minimize changes and avoid potentially expensive finalize() for gone callers. In this case, it means calling cbdataReferenceValid() instead of cbdataReferenceValidDone() here, but keeping the rest of the code intact (except for the _two_ new callBack() calls and no `cbdata` variable, of course).